src/cargo/core/features.rs

```diff
@@ -371,16 +372,33 @@ impl CliUnstable {
             );
         }
         for flag in flags {
-            self.add(flag)?;
+            let mut parts = flag.splitn(2, '=');
+            let k = parts.next().unwrap();
+            let v = parts.next();
+            self.add(k, v)?;
         }
         Ok(())
     }
 
-    fn add(&mut self, flag: &str) -> CargoResult<()> {
-        let mut parts = flag.splitn(2, '=');
-        let k = parts.next().unwrap();
-        let v = parts.next();
+    /// Read unstable options from a hashmap.
+    /// Intended for consuming unstable settings from config files
+    pub fn from_table(&mut self, flags: &HashMap<String, String>) -> CargoResult<()> {
```

Comment:
That'd be my preference too -- I didn't add it initially for fear of making too large an unsolicited diff, but I'll see what it looks like to swap over the little bespoke deserializer table.

Reply:
Hm, on closer reflection I think kicking this over to serde is orthogonal -- both the `parse` and `from_table` functions are updating-in-place, rather than deserializing to construct a new instance. I think that makes a difference here because we rely on being able to differentiate between "false because that's the default" and "explicitly set to false" to give the CLI the ability to override a config file in turning off a Zflag.